privacy/i2p.md

```diff
@@ -0,0 +1,77 @@
+Using I2P (Invisible Internet Project) with Qubes
+=================================================
+
+"The I2P network provides strong privacy protections for communication over the Internet. Many activities that would risk your privacy on the public Internet can be conducted anonymously inside I2P." - [I2P Website](https://geti2p.net/)
+
+
+Preparation - Basic
+===============
+
+The I2P Software will be running on a Linux-based TemplateVM. Here we are using Debian 9, but you could also use the default Debian 8 Template. So at first you need to install Debian 9 as a TemplateVM. If you want, you can create a dedicated clone of the default Debian-9 TemplateVM. In the Dom0 Terminal, run the command:
+~~~
+qvm-clone debian-9 debian-9-i2p
+~~~
+
+
+Installation
+============
+	
+1. Next, you need to add the I2P repository and the apt key to the new template. Therefore you need Internet access for your Template. If you have an active Internet connection, start your new TemplateVM, open the firewall configuration and allow full access for 5 minutes (this is neccessary for downloading the apt key for I2P). Now run the following commands (as root):
+	~~~
+	echo 'deb https://deb.i2p2.de/ stretch main' > /etc/apt/sources.list.d/i2p.list
+	echo 'deb-src https://deb.i2p2.de/ stretch main' >> /etc/apt/sources.list.d/i2p.list
+	wget https://geti2p.net/_static/i2p-debian-repo.key.asc
+	apt-key add i2p-debian-repo.key.asc
+	rm -rf i2p-debian-repo.key.asc
+	~~~
+	
+	If you are using Debian 8, replace 'stretch' with 'jessie' in the commands above.
+
+2. Then, update your package manager and install the i2p packages:
+	~~~
+	sudo apt update
+	sudo apt install i2p i2p-keyring
+	~~~
+	
+3. From your TemplateVM, you need to make the default i2p configuration directory persistent. This will allow I2P to save its configuration files from the AppVM which will be created later. Without this step, the I2P configuration would be lost when shutting down your AppVM. Now run these commands:
+	~~~
+	sudo su
+	mkdir -p /rw/config/qubes-bind-dirs.d
+	echo "binds+=( '/usr/share/i2p' )" > /rw/config/qubes-bind-dirs.d/50_user.conf
```

Comment:
This should be done in AppVM, not TemplateVM, or you should use `/etc/qubes-bind-dirs.d`. Your version will work on R3.2 only by a coincidence, because `/rw` is copied from TemplateVM to new AppVM during its creation. It is no longer true on R4.0.